pallets/utility/src/lib.rs

```diff
@@ -188,6 +190,10 @@ decl_module! {
         #[weight = batch_weight(&calls)]
         pub fn batch(origin, calls: Vec<<T as Trait>::Call>) {
             let is_root = ensure_root(origin.clone()).is_ok();
+            if !is_root {
+                let sender = ensure_signed(origin.clone())?;
+                CallPermissions::<T>::ensure_call_permissions(&sender)?;
+            }
             for (index, call) in calls.into_iter().enumerate() {
                 if let Err(e) = dispatch_call::<T>(origin.clone(), is_root, call) {
```

Comment:
The call name, and module name in  `pallets/permissions` should be updated for every tx dispatched in the batch.